### PR TITLE
Passing some UI Defaults to loleaflet frame

### PR DESCRIPTION
**Pull request by merttumer**
These ui defaults determine some widget states
at the start of the document which is hidden or shown
Some of them such as Ruler Statusbar and Sidebar can be
hidden at start with this patch. If not set, collabora
online will assume they are true

Signed-off-by: Mert Tumer <mert.tumer@collabora.com>

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] Sign-off message is added to all commits
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by merttumer** [comment]
@juliushaertl Can you please review these patches? Thank you

File: src/document.js [hunk 1]
```diff
@@ -80,6 +80,46 @@ const hideLoadingIndicator = () => {
...
+		'--color-loading-light': '--co-loading-light',
+		'--color-loading-dark': '--co-loading-dark',
+		'--color-box-shadow': '--co-box-shadow',
+		'--color-border': '--co-border',
+		'--color-border-dark': '--co-border-dark',
+		'--border-radius-pill': '--co-border-radius-pill'
+	}
+	var str = ''
+	for (var cssVarKey in cssVarMap) {
+		var cStyle = window.parent.getComputedStyle(document.documentElement).getPropertyValue(cssVarKey)
```
Comment (juliushaertl) [inline src/document.js:109]:
```suggestion
		var cStyle = window.getComputedStyle(document.documentElement).getPropertyValue(cssVarKey)
```

File: src/document.js [hunk 2]
```diff
@@ -80,6 +80,46 @@ const hideLoadingIndicator = () => {
...
+		if (!cStyle) {
+			// try suffix -dark instead
+			cStyle = window.parent.getComputedStyle(document.documentElement).getPropertyValue(cssVarKey + '-dark')
```
Comment (juliushaertl) [inline src/document.js:112]:
```suggestion
			cStyle = window.getComputedStyle(document.documentElement).getPropertyValue(cssVarKey + '-dark')
```

File: src/document.js [hunk 3]
```diff
@@ -176,7 +216,9 @@ const documentsMain = {
 
 			// form to post the access token for WOPISrc
 			const form = '<form id="loleafletform_viewer" name="loleafletform_viewer" target="loleafletframe_viewer" action="' + urlsrc + '" method="post">'
-				+ '<input name="access_token" value="' + accessToken + '" type="hidden"/></form>'
+				+ '<input name="access_token" value="' + accessToken + '" type="hidden"/>'
+				+ '<input name="ui_defaults" value="TextRuler=false;TextStatusbar=true;TextSidebar=false;PresentationSidebar=false;PresentationStatusbar=true;SpreadsheetSidebar=false" type="hidden"/'
```
Comment (juliushaertl) [inline src/document.js:220]:
```suggestion
				+ '<input name="ui_defaults" value="TextRuler=false;TextStatusbar=true;TextSidebar=false;PresentationSidebar=false;PresentationStatusbar=true;SpreadsheetSidebar=false" type="hidden"/>'
```

**Review by juliushaertl — CHANGES_REQUESTED** [review]
See inline for smaller comments. In general window.parent should be avoided as it doesn't work across different domains, which might occur in federated or global scale setups.

Otherwise :+1: from my side.

File: src/document.js (hunk 1 above)
Comment (juliushaertl) [reply src/document.js:109]: The CSS variables are also properly loaded inside of the document frame.

File: src/document.js (hunk 2 above)
Comment (merttumer) [reply src/document.js:112]: Thank you for the reply. Initially it was like this but later we discovered that it does not work on Firefox like that. The CSS variables become <empty string>. The only solution I found was using window.parent.

Comment (juliushaertl) [reply src/document.js:112]: Indeed seems to be a firefox bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1471231

Then let's get this in as it is and maybe see if we can work around that by not using `display: none;` but opacity of 0 until the frame is loaded.

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**